source/agora/common/EnrollmentManager.d

```diff
@@ -0,0 +1,353 @@
+/*******************************************************************************
+
+    Contains supporting code for enrollment process.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.EnrollmentManager;
+
+import agora.common.crypto.ECC;
+import agora.common.crypto.Key;
+import agora.common.crypto.Schnorr;
+import agora.common.crypto.Schnorr : Signature;
+import agora.common.Deserializer;
+import agora.common.Hash;
+import agora.common.Serializer;
+import agora.consensus.data.Enrollment;
+import agora.utils.Log;
+
+import d2sqlite3.database;
+import d2sqlite3.library;
+import d2sqlite3.results;
+import d2sqlite3.sqlite3;
+
+import std.algorithm;
+import std.file;
+import std.path;
+import std.stdio;
+
+mixin AddLogger!();
+
+/*******************************************************************************
+
+    Handle enrollment data and manage the validators set
+
+*******************************************************************************/
+
+public class EnrollmentManager
+{
+    /// SQLite db instance
+    private Database db;
+
+    /// Node's key pair
+    private KeyPair node_key_pair;
```

Comment:
```suggestion
    private Pair node_key_pair;
```

You could also loose the `node_` part if you want, as I don't think there's any other kind of key_pair the node will know of :)

Reply:
The node_key_pair must be KeyPair type because it is set from a node's keypair. This variable is set in the constructor, which will be called by a node later.

Reply:
He means to rename `node_key_pair` to `key_pair`

Reply:
His suggested change is about the type of the node_key_pair variable.

Reply:
You're right, I read it wrong.

Reply:
> The node_key_pair must be KeyPair type because it is set from a node's keypair.

But at the signature steps you turn it into a `Pair`.